modules/rest_api/controllers/services/rest.php

```diff
@@ -1637,54 +1642,21 @@ private function esGetSpecialFieldLatLon(array $doc) {
    * @return string
    *   Formatted value.
    */
-  private function esGetSpecialFieldLon(array $doc) {
+  private function esGetSpecialFieldLon(array $doc, array $params) {
     // Check in case fields are in composite agg key.
     $root = isset($doc['key']) ? $doc['key'] : $doc['location'];
     if (empty($root['point'])) {
       return 'n/a';
     }
     $coords = explode(',', $root['point']);
-    $ew = $coords[1] >= 0 ? 'E' : 'W';
-    $lon = number_format(abs($coords[1]), 3);
-    return "$lon$ew";
-  }
-
-  /**
-   * Special field handler for decimal latitude data.
-   *
-   * @param array $doc
-   *   Elasticsearch document.
-   *
-   * @return float
-   *   Decimal latitude.
-   */
-  private function esGetSpecialFieldDecLat(array $doc) {
-    // Check in case fields are in composite agg key.
-    $root = isset($doc['key']) ? $doc['key'] : $doc['location'];
-    if (empty($root['point'])) {
-      return 'n/a';
-    }
-    $coords = explode(',', $root['point']);
-    return $coords[0];
-  }
-
-  /**
-   * Special field handler for decimal longitude data.
-   *
-   * @param array $doc
-   *   Elasticsearch document.
-   *
-   * @return float
-   *   Decimal longitude.
-   */
-  private function esGetSpecialFieldDecLon(array $doc) {
-    // Check in case fields are in composite agg key.
-    $root = isset($doc['key']) ? $doc['key'] : $doc['location'];
-    if (empty($root['point'])) {
-      return 'n/a';
+    switch($params[0]) {
```

Comment:
Should we allow for a default when no options provided?
Also need to make sure docs are updated. Let me know if you want me to do that.

Reply:
Will do re default. I need to review docs in light of stuff I've recently done.